tensorflow/core/kernels/reduction_gpu_kernels.cu.h

```diff
@@ -652,64 +676,64 @@ void LaunchColumnReduction_LTE4096Cols(OpKernelContext* ctx, OUT_T out, IN_T in,
                                           sizeof(T) * extent_y * grid_dim.y)}),
                                       &temp_storage));
 
-    ColumnReduceKernel<<<grid_dim, block_dim, 0, cu_stream>>>(
+    GPU_LAUNCH_KERNEL(ColumnReduceKernel<IN_T, T*, Op>, grid_dim, block_dim, 0, cu_stream,
         in, (T*)temp_storage.flat<int8_t>().data(), extent_x, extent_y, op,
         init);
 
-    dim3 new_grid_dim((grid_dim.y * extent_y + 31) / 32, 1, 1);
+    dim3 new_grid_dim((grid_dim.y * extent_y + (TF_RED_WARPSIZE-1)) / TF_RED_WARPSIZE, 1, 1);
     dim3 num_threads(128, 1, 1);
-    CleanupSegments<<<new_grid_dim, block_dim, 0, cu_stream>>>(
-        (T*)temp_storage.flat<int8_t>().data(), out, extent_x, extent_y,
+     GPU_LAUNCH_KERNEL(CleanupSegments<T*,OUT_T,Op>, new_grid_dim, num_threads, 0, cu_stream,
+        ((T*)temp_storage.flat<int8_t>().data()), out, extent_x, extent_y,
         grid_dim.y, op, init);
   }
 }
 
 template <typename T, typename Op, typename OUT_T, typename IN_T>
 void LaunchColumnReduction(OpKernelContext* ctx, OUT_T out, IN_T in,
                            int extent_x, int extent_y, Op op, T init,
-                           const cudaStream_t& cu_stream) {
+                           const gpuStream_t& cu_stream) {
   if (extent_y <= 16) {
     LaunchColumnReduction_LTE16Cols(ctx, out, in, extent_x, extent_y, op, init,
                                     cu_stream);
   } else if (extent_y <= 4096) {
-    LaunchColumnReduction_LTE4096Cols(ctx, out, in, extent_x, extent_y, op,
+    LaunchColumnReduction_LTE4096Cols<T, Op, OUT_T, IN_T>(ctx, out, in, extent_x, extent_y, op,
```

Comment:
@iotamudelta it seems this is the only call path which leads to the crashing kernel? can we first check if the more native path, `ColumnReduceSimpleKernel`, work, perhaps by commenting out this path and mark it as FIXME?

Reply:
BTW from the look of the implementation `cu_stream` *ought* to be right. could you post links to IR / ISA dump here?

Reply:
my current suspicion is that the code object for the offending kernel might somehow be corrupted. so as an initial experiment let's try skip the path and see if we get the ball rolling.